### ISSUE TITLE
How to build only html version of documentation

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/AutoDoc#156]
This may be interesting in several context, e.g. publishing html elsewhere, or (my motivation) adding CI setup which avoids installing all needed LaTeX packages. I would assume that if HTML builds, then the generated XML file is correct, and that suits the purpose of AutoDoc CI tests.

**Comment by sebasguts** [comment]
AutoDoc only builds PDF manuals if there is a `pdflatex` executable in the path. Otherwise, it will only build HTML and text documentation. Is there a need of an explicit option for this?

**Comment by alex-konovalov** [comment]
Not urgently, but may be desirable, to ensure that anticipated tests will run in the same way on different machines, independently on whether pdflatex is there or not. But I can start without is, as it will not be on Travis by default, I guess...

**Comment by fingolfin** [comment]
Yes, this is highly desirable. But the problem with it is that GAPDoc does not really allow building just a HTML version, but @sebasguts has been working on that now, see https://github.com/frankluebeck/GAPDoc/pull/39

**Comment by sebasguts** [comment]
The functionality "kinda" exists. We could add an option to AutoDoc if that is desirable. Although a GAPDoc solution would be better I guess. See: https://github.com/gap-packages/AutoDoc/blob/master/gap/Magic.gi#L549

**Comment by fingolfin** [comment]
GAPDoc 1.6.3 added a "nopdf" option to `MakeGAPDocDoc`. So we could add a way to tell AutoDoc to pass that option for MakeGAPDocDoc (ideally with a check for the GAPDoc version, and a warning or error if GAPDoc is older than 1.6.3).

I think the most useful way for practical applications to do this is by checking `ValueOption("nopdf")` (and only that). This way, one can `PushOptions( rec( nopdf := true ) );` right before reading `makedoc.g` on a CI/CD server. And IMHO it shouldn't be a part of the regular `AutoDoc` options, because we do not want to encourage people to set `nopdf` per default in their `makedoc.g` files.

**Comment by alex-konovalov** [comment]
I can imagine a use of this in a CI test which builds the manual, just to check that it builds - markup errors will be caught, and the build will be quick since PDF is not needed.

**Comment by fingolfin** [comment]
Yeah, exactly. Or one might use it to generate HTML-only versions of a package manual for the master branch, to be put on a website (just like there are e.g. https://docs.julialang.org/en/v1.3-dev/ vs. https://docs.julialang.org/en/v1.1/ -- I'd love to have that for the GAP manual and GAP package manuals: access to all old versions of the manual, plus "latest" (i.e. latest released), and "dev".
